File: leetcode/rust/55.jump-game.rs
use std::cmp;

impl Solution {
    pub fn can_jump(nums: Vec<i32>) -> bool {
        let mut dis = 0;
        for i in 0..nums.len(){
          if dis >= i{
            dis = cmp::max(dis,(i + nums[i] as usize));
          }
      }
      if dis >= nums.len() -1
      {
        return true;
      }
      false
    }
}

File: leetcode/java/55.jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int d = nums[0];
        for (int i = 0; i < nums.length && i <= d; i++) {
            d = Math.max(d, i + nums[i]);
        }
        return d >= nums.length - 1;
    }
}
